eelsdbconverter/__init__.py

```diff
@@ -49,60 +48,186 @@ def __init__(self):
+        #Sample
+        sample = metadata.m_create(Sample)
+
+        sample.formula = file_data['formula']
+        sample.sample_id = str(file_data['id'])
+        sample.sample_title = file_data['title']
+
+        elements = file_data.get('elements')
         if elements is not None:
             if isinstance(elements, str):
                 elements = json.loads(elements)
-            material.atom_labels = elements
-        material.chemical_name = data.get('title')
-
-        experiment.m_create(
-            Method,
-            data_type='spectrum',
-            method_name='electron energy loss spectroscopy',
-            method_abbreviation='EELS')
-
-        section_data = experiment.m_create(Data)
-
-        archive.section_metadata.external_id = str(data.get('id'))
-        archive.section_metadata.external_db = 'EELSDB'
-
-        author = data.get('author')['name']
-        author = re.sub(r'\(.*\)', '', author).strip()
-        archive.section_metadata.coauthors = [Author(
-            first_name=' '.join(author.split(' ')[0:-1]),
-            last_name=author.split(' ')[-1])]
-        archive.section_metadata.comment = data.get('description')
-        archive.section_metadata.references = [
-            data.get('permalink'),
-            data.get('api_permalink'),
-            data.get('download_link')
-        ]
-
-        reference = data.get('reference')
-        if reference:
-            if isinstance(reference, str):
-                archive.section_metadata.references.append(reference)
-            elif isinstance(reference, dict):
-                if 'freetext' in reference:
-                    archive.section_metadata.references.append(
-                        re.sub(r'\r+\n+', '; ', reference['freetext']))
-                else:
-                    archive.section_metadata.references.append('; '.join([
-                        str(value).strip() for value in reference.values()
-                    ]))
+            sample.elements = elements
 
-        section_data.repository_name = 'Electron Energy Loss Spectroscopy (EELS) database'
-        section_data.repository_url = 'https://eelsdb.eu'
-        section_data.entry_repository_url = data.get('permalink')
+        #Experiment
+        experiment = metadata.m_create(Experiment)
+        experiment.method_name = 'electron energy loss spectroscopy'
+        experiment.method_abbreviation = 'EELS'
+        experiment.experiment_publish_time = datetime.strptime(
+            file_data.get('published'), '%Y-%m-%d %H:%M:%S')
+        edges = file_data.get('edges')
+        if edges is not None:
+            if isinstance(edges, str):
+                edges = json.loads(edges)
+            experiment.edges = edges
+
+        #Instrument
+        instrument = metadata.m_create(Instrument)
+        instrument.source_label = file_data['microscope']
+        device_settings = instrument.m_create(DeviceSettings)
+        device_settings.device_name = file_data['microscope']
```

Comment:
I am not sure how consistent all these keys are actually available. Running it over all of EELS will show.

Reply:
Yeah. These things were available in the two test cases that I was using. Is there an actionable step here for now?

Reply:
Yes, test the parser on the whole eelsdb. If it turns out that the data is not consistent, we have to make the array access safer.

Reply:
I pulled the entire eelsdb and resolved all the key bindings for the metadata files.
I, subsequently, found a few cases where the dataset is not consistent. I resolved the minor error with different separators in the files. But there are still a few problem cases.
These either don't have any header in the dataset file, or the units are undefined in the header of the dataset.

Should we do something about that?

Reply:
This is very the dirty business of data integration starts. Not having the data or having possibly flawed data. NOMAD usually does the second by carefully applying sensible defaults. E.g., you simply assume that it is eV.

Reply:
I can put that to be 'eV' in case of EELS Data, but there might be situations where it could be 'keV' or 'meV' and then we're off.

Reply:
I would still take eV as default and produce a warning via our logger.

Reply:
Alright, I've set it to eV by default. I'll add a logger warning.